### PR TITLE
[Storage] [DataMovement] [Blobs] Stress Scenarios

### PR DESCRIPTION
**Pull request by amnguye**
Creating all Stress Scenarios for Datamovement Blobs

(Apologies in advance for the long commit history, I was attempting to work in tangent when I had the last Stress Test PR open)

### CODE REVIEW

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobDirectoryScenarioBase.cs [hunk 1]
```diff
@@ -24,36 +24,39 @@ public BlobDirectoryUploadScenario(
...
+            StorageResource destinationResource = _blobsStorageResourceProvider.FromClient(
+                destinationContainerClient,
+                new()
+                {
+                    BlobDirectoryPrefix = pathPrefix,
+                    BlobType = new(blobType)
+                });
 
             // Initialize TransferManager
             TransferManager transferManager = new TransferManager();
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobDirectoryScenarioBase.cs:62]: Missed last time but this one is also extra. From here, you get the idea so probably just check all of them :)

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobSingleScenarioBase.cs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            {
+                BlockBlobClient destinationBlob = destinationContainerClient.GetBlockBlobClient(blobName);
+                destinationBaseBlob = destinationBlob;
+                destinationResource = _blobsStorageResourceProvider.FromClient(destinationBlob);
+            }
+
+            // Start Transfer
+            await new TransferValidator()
+            {
+                TransferManager = new()//new(_transferManagerOptions)
```
Comment (jalauzon-msft) [inline sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobSingleScenarioBase.cs:70]: Did you actually want to pass the options here? And below for `_dataTransferOptions`.

**Review by jalauzon-msft — APPROVED** [review]
Overall, looks good. Just some minor comments.

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Infrastructure/BlobTestSetupHelper.cs [hunk 3]
```diff
@@ -49,5 +73,46 @@ await blockBlobClient.UploadAsync(
...
         }
+
+        public static async Task CreateAppendBlobAsync(
+            AppendBlobClient appendBlobClient,
+            long? objectLength = Constants.KB * 4,
+            CancellationToken cancellationToken = default)
+        {
+            using Stream originalStream = await TestSetupHelper.CreateLimitedMemoryStream(objectLength.Value, cancellationToken: cancellationToken);
+            await appendBlobClient.CreateIfNotExistsAsync();
+            if (originalStream != default)
```
[the remark this answers is not in the document]
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Infrastructure/BlobTestSetupHelper.cs:84]: Yup not sure what I was thinking. I think this should have been if the object length was 0

[the remark this answers is not in the document]
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Infrastructure/BlobTestSetupHelper.cs]: done, applied it to the rest of the blob create helper methods

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/CopyBlobDirectoryScenarioBase.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            TransferManagerOptions transferManagerOptions,
+            DataTransferOptions dataTransferOptions,
+            TokenCredential sourceTokenCredential,
+            TokenCredential destinationTokenCredential,
+            Metrics metrics,
+            string testRunId)
+            : base(destinationBlobUri, blobSize, transferManagerOptions, dataTransferOptions, destinationTokenCredential, metrics, testRunId)
+        {
+            _sourceServiceClient = new BlobServiceClient(sourceBlobUri, sourceTokenCredential);
+            _blobCount = blobCount != default ? blobCount.Value : DataMovementBlobStressConstants.DefaultObjectCount;
```
[the remark this answers is not in the document]
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/CopyBlobDirectoryScenarioBase.cs]: removed

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Infrastructure/TestSetupHelper.cs [hunk 5]
```diff
@@ -72,14 +72,13 @@ public static async Task<StorageResource> GetTemporaryFileStorageResourceAsync(
...
             // so the StagedUploadAsync call is not prevented from reading using its FileStream.
-            Console.Out.Write($"Creating File: {localSourceFile}..");
             using (FileStream fileStream = File.Create(localSourceFile))
             {
-                Console.Out.WriteLine("Copying Stream to File..");
                 await originalStream.CopyToAsync(
                     fileStream,
                     bufferSize,
                     cancellationToken: cancellationToken);
+                fileStream.Close();
```
[the remark this answers is not in the document]
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Infrastructure/TestSetupHelper.cs]: removed

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/DownloadBlobDirectoryScenarioBase.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                cancellationToken);
+
+            // Create Destination Local Storage Resource
+            StorageResource sourceResource = await TestSetupHelper.GetTemporaryFileStorageResourceAsync(disposingLocalDirectory.DirectoryPath);
+
+            // Create Destination Storage Resource
+            StorageResource destinationResource = _blobsStorageResourceProvider.FromClient(sourceContainerClient, new() { BlobDirectoryPrefix = pathPrefix });
+
+            // Initialize TransferManager
+            TransferManager transferManager = new TransferManager();
```
[the remark this answers is not in the document]
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/DownloadBlobDirectoryScenarioBase.cs]: Yeah forgot to remove this. applied to the rest of the scenario bases

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/DownloadBlobSingleScenarioBase.cs [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+                    cancellationToken);
+                sourceBaseBlob = sourceBlob;
+                sourceResource = _blobsStorageResourceProvider.FromClient(sourceBlob);
+            }
+
+            // Create Local Destination Storage Resource
+            StorageResource destinationResource = _localFilesStorageResourceProvider.FromFile(Path.Combine(disposingLocalDirectory.DirectoryPath, blobName));
+
+            // Start Transfer
+            TransferManager transferManager = new TransferManager(_transferManagerOptions);
```
[the remark this answers is not in the document]
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/DownloadBlobSingleScenarioBase.cs]: removed

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobDirectoryScenarioBase.cs (hunk 1 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobDirectoryScenarioBase.cs]: removed

File: sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobSingleScenarioBase.cs (hunk 2 above)
Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement.Blobs/stress/src/Scenarios/UploadBlobSingleScenarioBase.cs]: yeah. I was testing something out and found a bug someplace else. added these options bag back in

**Review by Thompson1985 — APPROVED** [review]

**Pull request merged**